model/transaction/event.go

```diff
@@ -101,3 +120,33 @@ func (t *Event) Transform() common.MapStr {
 	}
 	return tx
 }
+
+func (e *Event) Transform(tctx *transform.Context) []beat.Event {
+	transformations.Inc()
+	events := []beat.Event{}
+	ev := beat.Event{
+		Fields: common.MapStr{
+			"processor":        processorTransEntry,
+			transactionDocType: e.fields(tctx),
+			"context":          tctx.Metadata.Merge(e.Context),
+		},
+		Timestamp: e.Timestamp,
+	}
+	events = append(events, ev)
+
+	spanCounter.Add(int64(len(e.Spans)))
+	for spIdx := 0; spIdx < len(e.Spans); spIdx++ {
+		sp := e.Spans[spIdx]
+		if sp.Timestamp.IsZero() {
```

Comment:
I'd move this to the `decode` method. In case the `span` relies on the timestamp for some calculation it would be nil.